pkg/apis/v1beta1/nodepool_validation_cel_test.go

```diff
@@ -97,4 +99,217 @@ var _ = Describe("CEL/Validation", func() {
 			Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
 		})
 	})
+	Context("KubeletConfiguration", func() {
+		It("should fail on kubeReserved with invalid keys", func() {
+			nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+				KubeReserved: v1.ResourceList{
+					v1.ResourcePods: resource.MustParse("2"),
+				},
+			}
+			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+		})
+		It("should fail on systemReserved with invalid keys", func() {
+			nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+				SystemReserved: v1.ResourceList{
+					v1.ResourcePods: resource.MustParse("2"),
+				},
+			}
+			Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+		})
+		Context("Eviction Signals", func() {
+			Context("Eviction Hard", func() {
+				It("should succeed on evictionHard with valid keys", func() {
+					nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+						EvictionHard: map[string]EvictionValue{
+							"memory.available":   "5%",
+							"nodefs.available":   "10%",
+							"nodefs.inodesFree":  "15%",
+							"imagefs.available":  "5%",
+							"imagefs.inodesFree": "5%",
+							"pid.available":      "5%",
+						},
+					}
+					Expect(env.Client.Create(ctx, nodePool)).To(Succeed())
+				})
+				It("should fail on evictionHard with invalid keys", func() {
+					nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+						EvictionHard: map[string]EvictionValue{
+							"memory": "5%",
+						},
+					}
+					Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+				})
+				It("should fail on invalid formatted percentage value in evictionHard", func() {
+					nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+						EvictionHard: map[string]EvictionValue{
+							"memory.available": "5%3",
+						},
+					}
+					Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+				})
+				It("should fail on invalid percentage value (too large) in evictionHard", func() {
+					nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+						EvictionHard: map[string]EvictionValue{
+							"memory.available": "110%",
+						},
+					}
+					Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+				})
+				It("should fail on invalid quantity value in evictionHard", func() {
+					nodePool.Spec.Template.Spec.Kubelet = &KubeletConfiguration{
+						EvictionHard: map[string]EvictionValue{
+							"memory.available": "110GB",
+						},
+					}
+					Expect(env.Client.Create(ctx, nodePool)).ToNot(Succeed())
+				})
+			})
+		})
+		Context("Eviction Soft", func() {
+			It("should succeed on evictionSoft with valid keys", func() {
```

Comment:
Should we try quantity values as well for success with the percentage values here?